File: push/src/main/java/com/xiaomi/push/service/ReconnectionManager.java
package com.xiaomi.push.service;

import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.stats.StatsHandler;
import com.xiaomi.xmsf.runtime.PushRuntime;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/service/ReconnectionManager.class */
class ReconnectionManager {
    private XMPushService mPushService;
    private PushReconnectState state = PushReconnectRuntime.initialState();

    public ReconnectionManager(XMPushService xMPushService) {
        this.mPushService = xMPushService;
    }

    public void onConnectSucceeded() {
        this.state = PushReconnectRuntime.onConnectSucceeded(System.currentTimeMillis());
        this.mPushService.removeJobs(1);
        PushRuntime.observeChannelEvent(null, "reconnect_succeeded", "ReconnectionManager.onConnectSucceeded");
    }

    public void tryReconnect(boolean z) {
        PushReconnectAttemptPlan planReconnect = PushReconnectRuntime.planReconnect(this.state, z, this.mPushService.shouldReconnect(), this.mPushService.hasJob(1), System.currentTimeMillis());
        this.state = planReconnect.getNextState();
        PushRuntime.observeChannelEvent(null, planReconnect.getEventAction(), "ReconnectionManager.tryReconnect");
        if (planReconnect.getAction() == PushReconnectAction.SkipNoReconnect) {
            MyLog.v("should not reconnect as no client or network.");
            return;
        }
        if (planReconnect.getAction() == PushReconnectAction.SkipExistingJob) {
            return;
        }
        if (planReconnect.getAction() == PushReconnectAction.Immediate) {
            this.mPushService.removeJobs(1);
            XMPushService xMPushService = this.mPushService;
            xMPushService.getClass();
            xMPushService.executeJob(new ConnectJob(xMPushService));
            return;
        }
        MyLog.w("schedule reconnect in " + planReconnect.getDelayMs() + "ms");
        XMPushService xMPushService2 = this.mPushService;
        xMPushService2.getClass();
        xMPushService2.executeJobDelayed(new ConnectJob(xMPushService2), (long) planReconnect.getDelayMs());
        if (planReconnect.getShouldDumpNativeNetInfo() && StatsHandler.getInstance().isAllowStats()) {
            NetworkCheckup.dumpNativeNetInfo();
        }
        if (planReconnect.getShouldRunConnectivityTest()) {
            NetworkCheckup.connectivityTest();
        }
    }
}

File: push/src/main/java/com/xiaomi/push/service/ReconnectionManager.kt
package com.xiaomi.push.service

import com.xiaomi.channel.commonutils.logger.MyLog
import com.xiaomi.stats.StatsHandler
import com.xiaomi.xmsf.runtime.PushRuntime

class ReconnectionManager(
    private val pushService: XMPushService,
) {
    private var state = PushReconnectRuntime.initialState()

    fun onConnectSucceeded() {
        state = PushReconnectRuntime.onConnectSucceeded(System.currentTimeMillis())
        pushService.removeJobs(1)
        PushRuntime.observeChannelEvent(null, "reconnect_succeeded", "ReconnectionManager.onConnectSucceeded")
    }

    fun tryReconnect(forceReconnect: Boolean) {
        val reconnectPlan = PushReconnectRuntime.planReconnect(
            state,
            forceReconnect,
            pushService.shouldReconnect(),
            pushService.hasJob(1),
            System.currentTimeMillis(),
        )
        state = reconnectPlan.nextState
        PushRuntime.observeChannelEvent(null, reconnectPlan.eventAction, "ReconnectionManager.tryReconnect")

        when (reconnectPlan.action) {
            PushReconnectAction.SkipNoReconnect -> {
                MyLog.v("should not reconnect as no client or network.")
                return
            }

            PushReconnectAction.SkipExistingJob -> return

            PushReconnectAction.Immediate -> {
                pushService.removeJobs(1)
                pushService.executeJob(ConnectJob(pushService))
                return
            }

            else -> Unit
        }

        MyLog.w("schedule reconnect in ${reconnectPlan.delayMs}ms")
        pushService.executeJobDelayed(ConnectJob(pushService), reconnectPlan.delayMs.toLong())
        if (reconnectPlan.shouldDumpNativeNetInfo && StatsHandler.getInstance().isAllowStats) {
            NetworkCheckup.dumpNativeNetInfo()
        }
        if (reconnectPlan.shouldRunConnectivityTest) {
            NetworkCheckup.connectivityTest()
        }
    }
}
